src/internal/bigbang/bigbang.go

```diff
@@ -0,0 +1,573 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package bigbang contains the logic for installing Big Bang and Flux
+package bigbang
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"slices"
+	"sort"
+	"strings"
+
+	"encoding/base64"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/defenseunicorns/pkg/helpers/v2"
+	fluxHelmCtrl "github.com/fluxcd/helm-controller/api/v2"
+	fluxSrcCtrl "github.com/fluxcd/source-controller/api/v1beta2"
+	"github.com/zarf-dev/zarf/src/api/v1alpha1"
+	"github.com/zarf-dev/zarf/src/config"
+	"github.com/zarf-dev/zarf/src/internal/packager/helm"
+	"github.com/zarf-dev/zarf/src/internal/packager/kustomize"
+	"github.com/zarf-dev/zarf/src/pkg/message"
+	"github.com/zarf-dev/zarf/src/pkg/utils"
+	"github.com/zarf-dev/zarf/src/pkg/variables"
+	"helm.sh/helm/v3/pkg/chartutil"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/yaml"
+)
+
+// Opts contains the options for the bigbang.Create function
+type Opts struct {
```

Comment:
Is my use of Opts here in line with what we want. I didn't call it BigBangOpts as revive will complain that it stutters, which is true. I also didn't feel like I should add these options to pkgConfig.